test/auth_scheme_with_extensions.test.ts

```diff
@@ -0,0 +1,162 @@
+                new Extension(2, new Uint8Array(0)),
+            ]);
+        }
+        const extensions = extensionsConstructor();
+        const serialized = extensions.serialize();
+        expect(serialized).toStrictEqual(new Uint8Array([0, 8, 0, 1, 0, 0, 0, 2, 0, 0]));
+    });
+
+    test('Repeated extension type', () => {
+        function extensionsConstructor() {
+            return new Extensions([
+                new Extension(1, new Uint8Array(0)),
+                new Extension(1, new Uint8Array(0)),
+            ]);
+        }
+        const extensions = extensionsConstructor();
+        const serialized = extensions.serialize();
+        expect(serialized).toStrictEqual(new Uint8Array([0, 8, 0, 1, 0, 0, 0, 1, 0, 0]));
+    });
+
+    test('Extensions 0 and MAX_UINT type', () => {
+        function extensionsConstructor() {
+            return new Extensions([
+                new Extension(0, new Uint8Array(0)),
+                new Extension(MAX_UINT16, new Uint8Array(0)),
+            ]);
+        }
+        const extensions = extensionsConstructor();
+        const serialized = extensions.serialize();
+        expect(serialized).toStrictEqual(new Uint8Array([0, 8, 0, 0, 0, 0, 255, 255, 0, 0]));
+    });
+});
+
+test.each(tokenVectors)('AuthScheme-TokenVector-%#', async (v: TokenVectors) => {
+    const tokenType = parseInt(v.token_type);
+    expect(tokenType).toBe(TOKEN_TYPES.PARTIALLY_BLIND_RSA.value);
+
+    const issuerName = hexToString(v.issuer_name);
+    const redemptionContext = hexToUint8(v.redemption_context);
+    const originInfo = v.origin_info !== '' ? hexToString(v.origin_info).split(',') : undefined;
+    const nonce = hexToUint8(v.nonce);
+    const keyId = hexToUint8(v.token_key_id);
+
+    const challenge = new TokenChallenge(tokenType, issuerName, redemptionContext, originInfo);
+    const challengeSerialized = challenge.serialize();
+    testSerialize(TokenChallenge, challenge);
+
+    const context = new Uint8Array(await crypto.subtle.digest('SHA-256', challengeSerialized));
+    // use PARTIALLY_BLIND_RSA because it is a has public extensions
+    const authInput = new AuthenticatorInput(
+        TOKEN_TYPES.PARTIALLY_BLIND_RSA,
+        TOKEN_TYPES.PARTIALLY_BLIND_RSA.value,
+        nonce,
+        context,
+        keyId,
+    );
+    const authInputEnc = authInput.serialize();
+
+    expect(uint8ToHex(authInputEnc)).toBe(v.token_authenticator_input);
+});
+
+describe('extensions', () => {
+    test('maxSizeExtension', () => {
+        const ext = new Extension(0xaa, new Uint8Array(Extension.MAX_EXTENSION_DATA_LENGTH));
+        testSerializeWithOps(Extension, ext);
+
+        const extBytes = ext.serialize();
+        expect(extBytes.length).toBe(2 + 2 + Extension.MAX_EXTENSION_DATA_LENGTH);
+    });
+
+    test('maxSizeArrayExtension', () => {
+        const ext = new Extension(0xaa, new Uint8Array(Extension.MAX_EXTENSION_DATA_LENGTH));
+        const arrayExt = new Extensions([ext]);
```

Comment:
with the current changes, an array can only have one maximum-length extension, but not two or more, for example, this throws an error

```
const ext1 = new Extension(0xaa, new Uint8Array(Extension.MAX_EXTENSION_DATA_LENGTH));
const ext2 = new Extension(0xbb new Uint8Array(Extension.MAX_EXTENSION_DATA_LENGTH));
const arrayExt = new Extensions([ext1, ext2]);
arrayExt.serialize();
```

If this is the intended behaviour, we should warn in comments.

Reply:
it is the expected behaviour yes.
I've added a comment.

Reply:
it is the expected behaviour yes.
I've added a comment.